src/ql/api/kernel.i

```diff
@@ -240,6 +240,19 @@ Returns
 None
 """
 
+%feature("doctring") ql::api::Kernel::state_prep
+"""
+Automatic state preparation, currently requires unitary decomposition for all cases.
+
+Parameters
+----------
+states : Complex Vector
+    Specifies the state (change) of the qubits after the state preparation
```

Comment:
Maybe asking a stupid question, but just to be clear: the state will be the same for all the "prepared" qubits right?

Reply:
The qubits will be in a state as specified by the vector, which (depending on the vector) can be something different for each of the qubits. (The state vector for |01> is [0 1 0 0]^T for example, so the first qubit is state |0> and the second is in state |1>) The state preparation takes the vector and uses it to prepare the total combined state on the specified qubits.

Reply:
Ah OK it's clearer now, thanks.